<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
        "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
    <title>TypeError: DirectorsViews.patch() got an unexpected keyword argument &#x27;did&#x27; // Werkzeug
        Debugger</title>
    <link rel="stylesheet" href="?__debugger__=yes&amp;cmd=resource&amp;f=style.css"
          type="text/css">
    <!-- We need to make sure this has a favicon so that the debugger does
         not accidentally trigger a request to /favicon.ico which might
         change the application's state. -->
    <link rel="shortcut icon"
          href="?__debugger__=yes&amp;cmd=resource&amp;f=console.png">
    <script src="?__debugger__=yes&amp;cmd=resource&amp;f=debugger.js"></script>
    <script type="text/javascript">
        var TRACEBACK = 2108563362176,
            CONSOLE_MODE = false,
            EVALEX = true,
            EVALEX_TRUSTED = false,
            SECRET = "1I6gZgAyEK4VIRMU21f7";
    </script>
</head>
<body style="background-color: #fff">
<div class="debugger">
    <h1>TypeError</h1>
    <div class="detail">
        <p class="errormsg">TypeError: DirectorsViews.patch() got an unexpected keyword argument &#x27;did&#x27;</p>
    </div>
    <h2 class="traceback">Traceback <em>(most recent call last)</em></h2>
    <div class="traceback">
        <h3></h3>
        <ul>
            <li>
                <div class="frame" id="frame-2108563361648">
                    <h4>File <cite class="filename">"C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py"</cite>,
                        line <em class="line">2091</em>,
                        in <code class="function">__call__</code></h4>
                    <div class="source ">
                        <pre class="line before"><span class="ws">    </span>def __call__(self, environ: dict, start_response: t.Callable) -&gt; t.Any:</pre>
                        <pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;The WSGI server calls the Flask application object as the</pre>
                        <pre class="line before"><span class="ws">        </span>WSGI application. This calls :meth:`wsgi_app`, which can be</pre>
                        <pre class="line before"><span class="ws">        </span>wrapped to apply middleware.</pre>
                        <pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;</pre>
                        <pre class="line current"><span class="ws">        </span>return self.wsgi_app(environ, start_response)</pre>
                    </div>
                </div>

            <li>
                <div class="frame" id="frame-2108563370144">
                    <h4>File <cite class="filename">"C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py"</cite>,
                        line <em class="line">2076</em>,
                        in <code class="function">wsgi_app</code></h4>
                    <div class="source ">
                        <pre class="line before"><span class="ws">            </span>try:</pre>
                        <pre class="line before"><span class="ws">                </span>ctx.push()</pre>
                        <pre class="line before"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
                        <pre class="line before"><span class="ws">            </span>except Exception as e:</pre>
                        <pre class="line before"><span class="ws">                </span>error = e</pre>
                        <pre class="line current"><span class="ws">                </span>response = self.handle_exception(e)</pre>
                        <pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
                        <pre class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre>
                        <pre class="line after"><span class="ws">                </span>raise</pre>
                        <pre class="line after"><span class="ws">            </span>return response(environ, start_response)</pre>
                        <pre class="line after"><span class="ws">        </span>finally:</pre>
                    </div>
                </div>

            <li>
                <div class="frame" id="frame-2108563358720">
                    <h4>File <cite class="filename">"C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py"</cite>,
                        line <em class="line">671</em>,
                        in <code class="function">error_router</code></h4>
                    <div class="source ">
                        <pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;</pre>
                        <pre class="line before"><span class="ws">        </span>if self._has_fr_route():</pre>
                        <pre class="line before"><span class="ws">            </span>try:</pre>
                        <pre class="line before"><span
                                class="ws">                </span>return self.handle_error(e)</pre>
                        <pre class="line before"><span class="ws">            </span>except Exception as f:</pre>
                        <pre class="line current"><span
                                class="ws">                </span>return original_handler(f)</pre>
                        <pre class="line after"><span class="ws">        </span>return original_handler(e)</pre>
                        <pre class="line after"><span class="ws"></span> </pre>
                        <pre class="line after"><span class="ws">    </span>def handle_error(self, e):</pre>
                        <pre class="line after"><span class="ws">        </span>&quot;&quot;&quot;</pre>
                        <pre class="line after"><span class="ws">        </span>Error handler for the API transforms a raised exception into a Flask response,</pre>
                    </div>
                </div>

            <li>
                <div class="frame" id="frame-2108563358624">
                    <h4>File <cite class="filename">"C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py"</cite>,
                        line <em class="line">669</em>,
                        in <code class="function">error_router</code></h4>
                    <div class="source ">
                        <pre class="line before"><span class="ws">        </span>:param function original_handler: the original Flask error handler for the app</pre>
                        <pre class="line before"><span class="ws">        </span>:param Exception e: the exception raised while handling the request</pre>
                        <pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;</pre>
                        <pre class="line before"><span class="ws">        </span>if self._has_fr_route():</pre>
                        <pre class="line before"><span class="ws">            </span>try:</pre>
                        <pre class="line current"><span
                                class="ws">                </span>return self.handle_error(e)</pre>
                        <pre class="line after"><span class="ws">            </span>except Exception as f:</pre>
                        <pre class="line after"><span class="ws">                </span>return original_handler(f)</pre>
                        <pre class="line after"><span class="ws">        </span>return original_handler(e)</pre>
                        <pre class="line after"><span class="ws"></span> </pre>
                        <pre class="line after"><span class="ws">    </span>def handle_error(self, e):</pre>
                    </div>
                </div>

            <li>
                <div class="frame" id="frame-2108563361744">
                    <h4>File <cite class="filename">"C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py"</cite>,
                        line <em class="line">2073</em>,
                        in <code class="function">wsgi_app</code></h4>
                    <div class="source ">
                        <pre class="line before"><span
                                class="ws">        </span>ctx = self.request_context(environ)</pre>
                        <pre class="line before"><span class="ws">        </span>error: t.Optional[BaseException] = None</pre>
                        <pre class="line before"><span class="ws">        </span>try:</pre>
                        <pre class="line before"><span class="ws">            </span>try:</pre>
                        <pre class="line before"><span class="ws">                </span>ctx.push()</pre>
                        <pre class="line current"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
                        <pre class="line after"><span class="ws">            </span>except Exception as e:</pre>
                        <pre class="line after"><span class="ws">                </span>error = e</pre>
                        <pre class="line after"><span class="ws">                </span>response = self.handle_exception(e)</pre>
                        <pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
                        <pre class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre>
                    </div>
                </div>

            <li>
                <div class="frame" id="frame-2108563373072">
                    <h4>File <cite class="filename">"C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py"</cite>,
                        line <em class="line">1518</em>,
                        in <code class="function">full_dispatch_request</code></h4>
                    <div class="source ">
                        <pre class="line before"><span class="ws">            </span>request_started.send(self)</pre>
                        <pre class="line before"><span
                                class="ws">            </span>rv = self.preprocess_request()</pre>
                        <pre class="line before"><span class="ws">            </span>if rv is None:</pre>
                        <pre class="line before"><span
                                class="ws">                </span>rv = self.dispatch_request()</pre>
                        <pre class="line before"><span class="ws">        </span>except Exception as e:</pre>
                        <pre class="line current"><span class="ws">            </span>rv = self.handle_user_exception(e)</pre>
                        <pre class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
                        <pre class="line after"><span class="ws"></span> </pre>
                        <pre class="line after"><span class="ws">    </span>def finalize_request(</pre>
                        <pre class="line after"><span class="ws">        </span>self,</pre>
                        <pre class="line after"><span class="ws">        </span>rv: t.Union[ResponseReturnValue, HTTPException],</pre>
                    </div>
                </div>

            <li>
                <div class="frame" id="frame-2108563373024">
                    <h4>File <cite class="filename">"C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py"</cite>,
                        line <em class="line">671</em>,
                        in <code class="function">error_router</code></h4>
                    <div class="source ">
                        <pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;</pre>
                        <pre class="line before"><span class="ws">        </span>if self._has_fr_route():</pre>
                        <pre class="line before"><span class="ws">            </span>try:</pre>
                        <pre class="line before"><span
                                class="ws">                </span>return self.handle_error(e)</pre>
                        <pre class="line before"><span class="ws">            </span>except Exception as f:</pre>
                        <pre class="line current"><span
                                class="ws">                </span>return original_handler(f)</pre>
                        <pre class="line after"><span class="ws">        </span>return original_handler(e)</pre>
                        <pre class="line after"><span class="ws"></span> </pre>
                        <pre class="line after"><span class="ws">    </span>def handle_error(self, e):</pre>
                        <pre class="line after"><span class="ws">        </span>&quot;&quot;&quot;</pre>
                        <pre class="line after"><span class="ws">        </span>Error handler for the API transforms a raised exception into a Flask response,</pre>
                    </div>
                </div>

            <li>
                <div class="frame" id="frame-2108563372976">
                    <h4>File <cite class="filename">"C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py"</cite>,
                        line <em class="line">669</em>,
                        in <code class="function">error_router</code></h4>
                    <div class="source ">
                        <pre class="line before"><span class="ws">        </span>:param function original_handler: the original Flask error handler for the app</pre>
                        <pre class="line before"><span class="ws">        </span>:param Exception e: the exception raised while handling the request</pre>
                        <pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;</pre>
                        <pre class="line before"><span class="ws">        </span>if self._has_fr_route():</pre>
                        <pre class="line before"><span class="ws">            </span>try:</pre>
                        <pre class="line current"><span
                                class="ws">                </span>return self.handle_error(e)</pre>
                        <pre class="line after"><span class="ws">            </span>except Exception as f:</pre>
                        <pre class="line after"><span class="ws">                </span>return original_handler(f)</pre>
                        <pre class="line after"><span class="ws">        </span>return original_handler(e)</pre>
                        <pre class="line after"><span class="ws"></span> </pre>
                        <pre class="line after"><span class="ws">    </span>def handle_error(self, e):</pre>
                    </div>
                </div>

            <li>
                <div class="frame" id="frame-2108563372928">
                    <h4>File <cite class="filename">"C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py"</cite>,
                        line <em class="line">1516</em>,
                        in <code class="function">full_dispatch_request</code></h4>
                    <div class="source ">
                        <pre class="line before"><span class="ws">        </span>self.try_trigger_before_first_request_functions()</pre>
                        <pre class="line before"><span class="ws">        </span>try:</pre>
                        <pre class="line before"><span class="ws">            </span>request_started.send(self)</pre>
                        <pre class="line before"><span
                                class="ws">            </span>rv = self.preprocess_request()</pre>
                        <pre class="line before"><span class="ws">            </span>if rv is None:</pre>
                        <pre class="line current"><span
                                class="ws">                </span>rv = self.dispatch_request()</pre>
                        <pre class="line after"><span class="ws">        </span>except Exception as e:</pre>
                        <pre class="line after"><span
                                class="ws">            </span>rv = self.handle_user_exception(e)</pre>
                        <pre class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
                        <pre class="line after"><span class="ws"></span> </pre>
                        <pre class="line after"><span class="ws">    </span>def finalize_request(</pre>
                    </div>
                </div>

            <li>
                <div class="frame" id="frame-2108563372880">
                    <h4>File <cite class="filename">"C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py"</cite>,
                        line <em class="line">1502</em>,
                        in <code class="function">dispatch_request</code></h4>
                    <div class="source ">
                        <pre class="line before"><span class="ws">            </span>getattr(rule, &quot;provide_automatic_options&quot;, False)</pre>
                        <pre class="line before"><span class="ws">            </span>and req.method == &quot;OPTIONS&quot;</pre>
                        <pre class="line before"><span class="ws">        </span>):</pre>
                        <pre class="line before"><span class="ws">            </span>return self.make_default_options_response()</pre>
                        <pre class="line before"><span class="ws">        </span># otherwise dispatch to the handler for that endpoint</pre>
                        <pre class="line current"><span class="ws">        </span>return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)</pre>
                        <pre class="line after"><span class="ws"></span> </pre>
                        <pre class="line after"><span class="ws">    </span>def full_dispatch_request(self) -&gt; Response:</pre>
                        <pre class="line after"><span class="ws">        </span>&quot;&quot;&quot;Dispatches the request and on top of that performs request</pre>
                        <pre class="line after"><span class="ws">        </span>pre and postprocessing as well as HTTP exception catching and</pre>
                        <pre class="line after"><span class="ws">        </span>error handling.</pre>
                    </div>
                </div>

            <li>
                <div class="frame" id="frame-2108563372832">
                    <h4>File <cite class="filename">"C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py"</cite>,
                        line <em class="line">403</em>,
                        in <code class="function">wrapper</code></h4>
                    <div class="source ">
                        <pre class="line before"><span class="ws">        </span>:param resource: The resource as a flask view function</pre>
                        <pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;</pre>
                        <pre class="line before"><span class="ws"></span> </pre>
                        <pre class="line before"><span class="ws">        </span>@wraps(resource)</pre>
                        <pre class="line before"><span class="ws">        </span>def wrapper(*args, **kwargs):</pre>
                        <pre class="line current"><span
                                class="ws">            </span>resp = resource(*args, **kwargs)</pre>
                        <pre class="line after"><span
                                class="ws">            </span>if isinstance(resp, BaseResponse):</pre>
                        <pre class="line after"><span class="ws">                </span>return resp</pre>
                        <pre class="line after"><span
                                class="ws">            </span>data, code, headers = unpack(resp)</pre>
                        <pre class="line after"><span class="ws">            </span>return self.make_response(data, code, headers=headers)</pre>
                        <pre class="line after"><span class="ws"></span> </pre>
                    </div>
                </div>

            <li>
                <div class="frame" id="frame-2108563372784">
                    <h4>File <cite class="filename">"C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\views.py"</cite>,
                        line <em class="line">84</em>,
                        in <code class="function">view</code></h4>
                    <div class="source ">
                        <pre class="line before"><span class="ws">        </span>constructor of the class.</pre>
                        <pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;</pre>
                        <pre class="line before"><span class="ws"></span> </pre>
                        <pre class="line before"><span class="ws">        </span>def view(*args: t.Any, **kwargs: t.Any) -&gt; ResponseReturnValue:</pre>
                        <pre class="line before"><span class="ws">            </span>self = view.view_class(*class_args, **class_kwargs)  # type: ignore</pre>
                        <pre class="line current"><span class="ws">            </span>return current_app.ensure_sync(self.dispatch_request)(*args, **kwargs)</pre>
                        <pre class="line after"><span class="ws"></span> </pre>
                        <pre class="line after"><span class="ws">        </span>if cls.decorators:</pre>
                        <pre class="line after"><span class="ws">            </span>view.__name__ = name</pre>
                        <pre class="line after"><span
                                class="ws">            </span>view.__module__ = cls.__module__</pre>
                        <pre class="line after"><span
                                class="ws">            </span>for decorator in cls.decorators:</pre>
                    </div>
                </div>

            <li>
                <div class="frame" id="frame-2108563372736">
                    <h4>File <cite class="filename">"C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\resource.py"</cite>,
                        line <em class="line">49</em>,
                        in <code class="function">dispatch_request</code></h4>
                    <div class="source ">
                        <pre class="line before"><span class="ws">        </span>for decorator in self.method_decorators:</pre>
                        <pre class="line before"><span class="ws">            </span>meth = decorator(meth)</pre>
                        <pre class="line before"><span class="ws"></span> </pre>
                        <pre class="line before"><span class="ws">        </span>self.validate_payload(meth)</pre>
                        <pre class="line before"><span class="ws"></span> </pre>
                        <pre class="line current"><span class="ws">        </span>resp = meth(*args, **kwargs)</pre>
                        <pre class="line after"><span class="ws"></span> </pre>
                        <pre class="line after"><span class="ws">        </span>if isinstance(resp, BaseResponse):</pre>
                        <pre class="line after"><span class="ws">            </span>return resp</pre>
                        <pre class="line after"><span class="ws"></span> </pre>
                        <pre class="line after"><span class="ws">        </span>representations = self.representations or {}</pre>
                    </div>
                </div>
        </ul>
        <blockquote>TypeError: DirectorsViews.patch() got an unexpected keyword argument &#x27;did&#x27;</blockquote>
    </div>

    <div class="plain">
        <p>
            This is the Copy/Paste friendly version of the traceback.
        </p>
        <textarea cols="50" rows="10" name="code" readonly>Traceback (most recent call last):
  File &quot;C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py&quot;, line 2091, in __call__
    return self.wsgi_app(environ, start_response)
  File &quot;C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py&quot;, line 2076, in wsgi_app
    response = self.handle_exception(e)
  File &quot;C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py&quot;, line 671, in error_router
    return original_handler(f)
  File &quot;C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py&quot;, line 669, in error_router
    return self.handle_error(e)
  File &quot;C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py&quot;, line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File &quot;C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py&quot;, line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &quot;C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py&quot;, line 671, in error_router
    return original_handler(f)
  File &quot;C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py&quot;, line 669, in error_router
    return self.handle_error(e)
  File &quot;C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py&quot;, line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File &quot;C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py&quot;, line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File &quot;C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py&quot;, line 403, in wrapper
    resp = resource(*args, **kwargs)
  File &quot;C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\views.py&quot;, line 84, in view
    return current_app.ensure_sync(self.dispatch_request)(*args, **kwargs)
  File &quot;C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\resource.py&quot;, line 49, in dispatch_request
    resp = meth(*args, **kwargs)
TypeError: DirectorsViews.patch() got an unexpected keyword argument &#x27;did&#x27;</textarea>
    </div>
    <div class="explanation">
        The debugger caught an exception in your WSGI application. You can now
        look at the traceback which led to the error. <span class="nojavascript">
  If you enable JavaScript you can also use additional features such as code
  execution (if the evalex feature is enabled), automatic pasting of the
  exceptions and much more.</span>
    </div>
    <div class="footer">
        Brought to you by <strong class="arthur">DON'T PANIC</strong>, your
        friendly Werkzeug powered traceback interpreter.
    </div>
</div>

<div class="pin-prompt">
    <div class="inner">
        <h3>Console Locked</h3>
        <p>
            The console is locked and needs to be unlocked by entering the PIN.
            You can find the PIN printed out on the standard output of your
            shell that runs the server.
        <form>
            <p>PIN:
                <input type=text name=pin size=14>
                <input type=submit name=btn value="Confirm Pin">
        </form>
    </div>
</div>
</body>
</html>

<!--

Traceback (most recent call last):
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py", line 2091, in __call__
    return self.wsgi_app(environ, start_response)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py", line 2076, in wsgi_app
    response = self.handle_exception(e)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py", line 671, in error_router
    return original_handler(f)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py", line 669, in error_router
    return self.handle_error(e)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py", line 671, in error_router
    return original_handler(f)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py", line 669, in error_router
    return self.handle_error(e)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\api.py", line 403, in wrapper
    resp = resource(*args, **kwargs)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask\views.py", line 84, in view
    return current_app.ensure_sync(self.dispatch_request)(*args, **kwargs)
  File "C:\Users\user\AppData\Local\Programs\Python\Python310\Lib\site-packages\flask_restx\resource.py", line 49, in dispatch_request
    resp = meth(*args, **kwargs)
TypeError: DirectorsViews.patch() got an unexpected keyword argument 'did'

-->
